platform/default/http_request_curl.cpp

```diff
@@ -75,7 +75,7 @@ class HTTPCURLContext : public HTTPContext<HTTPCURLContext> {
 
 public:
     // Used as the CURL timer function to periodically check for socket updates.
-    uv_timer_t *timeout = nullptr;
+    uv_timer_t timeout;
```

Comment:
We should keep this as a pointer since we need to call `uv::close` to properly remove it from the loop handle later. When the destructor runs, we still have to wait for the loop to call our close callback until we can actually delete this timer object.